src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -606,344 +610,164 @@ public void takeItemStackFromProvider(@NotNull final ICapabilityProvider provide
     }
 
     /**
-     * Wait for a needed shovel.
+     * Wait for a needed tool or weapon.
      *
      * @return NEEDS_SHOVEL
      */
     @NotNull
-    private AIState waitForShovel()
+    private AIState waitForToolOrWeapon()
     {
-        if (checkForShovel())
+        final IToolType toolType = worker.getWorkBuilding().getNeedsTool();
+        if (toolType != ToolType.NONE)
         {
-            delay += DELAY_RECHECK;
-            return NEEDS_SHOVEL;
+            if (checkForToolOrWeapon(toolType))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)